### ISSUE TITLE
Crash when reading directory

### ISSUE THREAD

**Issue by tspivey** [relan/exfat#146]
Steps to reproduce:
```
dd if=/dev/zero of=exfat.img bs=1024k count=10
mkfs.exfat exfat.img
mount exfat.img /mnt
touch /mnt/$'\355''a '
ls /mnt
```

After running that, I get:
```ls: reading directory '/mnt': Software caused connection abort```

**Comment by relan** [comment]
Thanks for the detailed bug report!

The root cause is the utf8_to_wchar() function: is does not check the upper 2 bits of the 2nd and subsequent bytes in the UTF-8 sequence.

**Comment by relan** [comment]
Fixed in 66e3a255147d1714f313399bfdfb52fff8afbf50.
